Implementation of my portfolio's mobile version based on a Figma template

rskinnerc opened the pull request:
### In this project, I set up and built the first mobile version of my portfolio based on a Microvere's Figma template.

#### In this PR I have:
- Installed and configured the linters: Webhint and Stylelint
- Created a .gitignore file and added the node_modules folder
- Added the .github/workflows folder and the provided linters.yml file
- Added the index.html file and created the structure according to the template
- Added the styles.css file and linked it in the index.html
- Added the background, social networks, menu and caret images
- Added the required styles to mimic the layout using flexbox and the rest of the styles as similar as possible
- Added a README.md file using the provided template and updated it accordingly

Commit:
set github workflows config file

Commit:
init node pkg and add node_modules to gitignore

Commit:
install and configure webhint

Commit:
install and configure stylelint linter

Commit:
add index.html file and basic html structure

Commit:
add styles.css file and link it in index.html

Commit:
add the navbar content and styles

Add the menu icon
Style the navbar using flexbox

Commit:
add headline title and supporting text

Structure the headline content using flexbox
Style the title and supporting text accordingly

Commit:
fix units for paddingand margin when its 0

Commit:
fix logo color

Commit:
refactor the README.md adding details

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+  <nav class="toolbar">
+    <span class="logo">Welcome</span>
+    <img src="./Enabled.png" alt="Manu Icon" class="menu-icon">
+  </nav>
```
- [x] It's a good idea to use the right semantic tag on the page whenever possible. Please wrap the `toolbar` section in a `header` tag and the hamburger icon with a `nav` tag. 
You can read more on semantic tag [here](https://www.w3schools.com/html/html5_semantic_elements.asp)

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+
+<body>
+  <nav class="toolbar">
+    <span class="logo">Welcome</span>
```
- [x] In this case, this element is supposed to be the logo. kindly use an `<a>` tag to wrap the logo instead of a `span` tag.

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+      don’t hestiate to contact me.</p>
+    <img src="./social-media-icons.png" alt="Social Media Menu" class="social-media">
+    <img src="./Disabled.png" alt="More" class="more-button">
+  </header>
```
- [x] Kindly wrap the `Headline` section in a `<section>` tag instead of a `<header>` tag.

You should only use the `<header>` element to identify content that precedes the primary content of the web page and it often contains website branding, navigation elements, search forms, and similar content that is duplicated across all or most pages of a website.

Read more: https://html.com/tags/header/#ixzz7NkefLEt8

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+    <img src="./Enabled.png" alt="Manu Icon" class="menu-icon">
+  </nav>
+  <header class="headline">
+    <span class="title naval">Hey There. I'm Ronald</span>
```
- [x] Kindly wrap the title in a `<h1>` tag

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+  </nav>
+  <header class="headline">
+    <span class="title naval">Hey There. I'm Ronald</span>
+    <span class="title green">I'm a software developer</span>
```
- [x] Kindly wrap the subtitle in a `h2` or `h3`...`h6` tag

phelian23 commented on styles.css:
```diff
@@ -0,0 +1,62 @@
+}
+
+.naval {
+  color: #172b4d;
```
- [x] Instead of using `font-family: Inter, sans-serif;` multiple times, kindly create a class say
``` css
.inter {
  font-family: Inter, sans-serif;
}
```
as this will help your code to follow the `DRY` rule.

phelian23 commented on README.md:
```diff
@@ -1,2 +1,40 @@
+- HTML
+- CSS
+- Code validation using Webhint, Stylelint and Lighthouse
+
```
- [x] For the sake of people that don't have enough technical knowledge, I suggest you Kindly add detailed installation guidelines to your README FILE. e.g;

### Installation

If you installed git you can clone the code to your machine, or download a ZIP of all the files directly.

[Download the ZIP from this location](https://github.com/), or run the following [git](https://git-scm.com/downloads) command to clone the files to your machine:

`git clone https://github.com/rskinnerc/portafolio.git`
Once the files are on your machine, open the `portafolio` folder in [Visual Studio Code](https://code.visualstudio.com/).

With the files open in Visual Studio Code, press the Go Live button at the bottom of the window to launch the files with [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer).

phelian23 commented on index.html:
```diff
@@ -0,0 +1,30 @@
+      experience! If you like
+      what you see and have a project you need coded,
+      don’t hestiate to contact me.</p>
+    <img src="./social-media-icons.png" alt="Social Media Menu" class="social-media">
```
- [x] Kindly add the social media icons individually as a list, so that links can be added to each to direct it to the right path

phelian23 reviewed (changes requested):
Hi @rskinnerc, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- All linter checks are passing ✔️
- Descriptive commits and correct use of the Git Flow 🔥
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Commit:
apply the required changes

Improve html structure using semantic tags
Refactor font styles sto avoid repetition
Refactor social media icons to make them individual
Improve README.md to explain how to install and preview the repository

Commit:
fix styles.css file

ShadyShawkat commented on index.html:
```diff
@@ -10,21 +10,30 @@
-    <span class="title green">I'm a software developer</span>
-    <p class="supporting-text naval"> I can help you build a product , feature or website Look through some of my work
+  <header class="toolbar">
+    <a href="/" class="logo inter">Welcome</a>
```
- [x] Kindly adjust the design of the logo to look like the Figma design. (You can use the `text-decoration` property)

rskinnerc replied:
Thank you, I'm sorry I forgot that the default decoration for links was underlined. I already fixed it

ShadyShawkat commented on index.html:
```diff
@@ -10,21 +10,30 @@
       what you see and have a project you need coded,
       don’t hestiate to contact me.</p>
-    <img src="./social-media-icons.png" alt="Social Media Menu" class="social-media">
+    <div class="social-media">
+      <img src="./twitter.svg" alt="Follow me in Twitter">
+      <img src="./linkedin-icon.svg" alt="Follow me in LinkedIn">
+      <img src="./medium.svg" alt="Medium">
+      <img src="./github.svg" alt="My Github">
+      <img src="./angellist.svg" alt="Angellist">
+    </div>
```
- [x] Please wrap make this into a list using `ul` and `li` tags which is in the requirement and is much more semantic.
```
ul
   li
      img
   /li
   li
      img
   /li
   ...
/ul
```

rskinnerc replied:
Although it was not specifically stated in the requirements, I really appreciate your recommendation. Clearly seems more semantic Thank you

ShadyShawkat commented on index.html:
```diff
@@ -10,21 +10,30 @@
       what you see and have a project you need coded,
       don’t hestiate to contact me.</p>
-    <img src="./social-media-icons.png" alt="Social Media Menu" class="social-media">
+    <div class="social-media">
+      <img src="./twitter.svg" alt="Follow me in Twitter">
+      <img src="./linkedin-icon.svg" alt="Follow me in LinkedIn">
+      <img src="./medium.svg" alt="Medium">
+      <img src="./github.svg" alt="My Github">
+      <img src="./angellist.svg" alt="Angellist">
+    </div>
     <img src="./Disabled.png" alt="More" class="more-button">
```
- [x] Kindly wrap this in `ul` and `li` tags instead of `div` tag

ShadyShawkat commented on index.html:
```diff
@@ -10,21 +10,30 @@
-    <span class="title green">I'm a software developer</span>
-    <p class="supporting-text naval"> I can help you build a product , feature or website Look through some of my work
+  <header class="toolbar">
+    <a href="/" class="logo inter">Welcome</a>
```
- [x] Please change the design of the logo to match the Figma design.

ShadyShawkat reviewed (changes requested):
## Hi @rskinnerc, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![Image](https://c.tenor.com/CWrSBML0hC8AAAAC/little-dinosaur-so-close.gif)

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AnselemOdims commented on styles.css:
```diff
@@ -0,0 +1,71 @@
+  margin: 60px 0;
+}
+
+.social-media {
+  width: 170px;
+  height: 20px;
+  align-self: center;
+  display: flex;
+  justify-content: space-evenly;
```
- [x] [OPTIONAL] Kindly remove the list-style from the unordered list. I am including it as optional because the previous reviewer did not point it out but I strongly suggest you implement it. You can use `list-style: none`

rskinnerc replied:
Thank you I'll do

AnselemOdims reviewed (approved):
### Hi  @rskinnerc 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
